037-kafka-canary-roadmap.md

```diff
@@ -0,0 +1,127 @@
+
+* Support for SASL `OAUTHBEARER` mechanism. Sarama supports this.
+* Support for forcing leader elections. Sarama doesn't support this.
+* Support for transactional produce. Sarama doesn't seem to support this, at least not in a usable way. You can send the transaction-related requests, but there is no client library support akin to the `TransactionManager`, for ensuring the RPCs are sent correctly. 
+
+All of the desirable features will be implemented after Canary re-implementation.
+
+### Differential observations
+
+By running multiple producers and consumers, Canary could more accurately indicate where problems lay. 
+
+For example, in a cluster with multiple listeners using different authentication mechanisms, if produce requests fail using an `SASL/OAUTHBEARER`-authenticated listener but succeed using an`SASL/PLAIN`-authenticated listener, you can infer that the problem is with the login mechanism. 
+
+There are many dimensions in which you can run multiple producers and consumers: 
+
+* DNS resolution
+* TLS (e.g. different versions or detecting imminent broker certificate expiry)
+* Producing to replicas on different volumes
+* With/without transactional producers, to flush out problems with transaction coordinators.
+* Similarly with/without using consumer groups. Using different group ids to detect problems with group coordinators.
+
+
+### Actionable documentation
+
+Assuming the differential observations feature proves to provide worthwhile information, we would proceed to also provide documented procedures recommending corrective actions based on the specific faults it reported.
+
+### Changes and removals
+
+* As logging mechanism will be used Log4j2
+  * The log levels will have to be changed from `0, 1, 2` to `INFO, DEBUG, WARN, TRACE`.
+  * `VERBOSITY_LOG_LEVEL` will be changed to `CANARY_LOG_LEVEL` and used inside the `log4j2.properties` file - default will be `INFO`
+* `SARAMA_LOG_ENABLED` environment variable will be removed
+* Implementation of dynamic watcher will be removed 
+  * For logging changes the dynamic reconfiguration will be used from Log4j2 (similarly to operators repository)
+
+## Affected/not affected projects
+
+This affects the Canary project only.
+
+## Compatibility
+
+The API of the existing Canary comprises the following:
+
+* The configuration env vars
+* Metrics (via the Prometheus scrape endpoint)
+* The status and health endpoints
+
+### Configuration
+
+Java clients are normally configured via `.properties` files.
+Providing a compatibility mapping from env var to properties would be a lot of work, given that the Canary hasn't seen very wide usage.
+Some options would become obselete (e.g. `KAFKA_VERSION`), others are incompatible with how the Java clients work (e.g. `KAFKA_BOOTSTRAP_BACKOFF_MAX_ATTEMPTS`).
+While we can strive to minimize incompatibilities, it seems of questionable value.
+So it seems better to provide a clean break.
```

Comment:
This is fine. But you should also explain how will the configuration look like (perhps not here but in a separate part).